python/paddle/fluid/layers/nn.py

```diff
@@ -5354,3 +5355,70 @@ def rank_loss(label, left, right, name=None):
                 "Right": right},
         outputs={'Out': out})
     return out
+
+
+def flatten(x, axis=1, name=None):
+    """
+    **Flatten layer**
+    Flattens the input tensor into a 2D matrix.
+
+    Examples:
+    Case 1:
+      Given
+        X.shape = (3, 100, 100, 4)
+      and
+        axis = 2
+      We get:
+        Out.shape = (3 * 100, 4 * 100)
```

Comment:
why is this not (3, 100, 4 * 100)?

Reply:
output is a 2D tensor,`(d_0 X d_1 ... d_(axis-1), d_axis X d_(axis+1) ... X dn)`